### ISSUE TITLE
Problems of Visdom after removing GAN loss in pix2pix model

### ISSUE THREAD

**Issue by wl082013** [junyanz/pytorch-CycleGAN-and-pix2pix#1216]
Hi, 
I removed the GAN loss and only used the  L1 loss to see the effect of training.
In this case, I remove the discriminator and its relevant optimizers and backpropagation functions.
However, when I started training, there is an unexpected bug in the visualizer. plot_current_losses:

 if not hasattr(self, 'plot_data'):
            self.plot_data = {'X': [], 'Y': [], 'legend': list(losses.keys())}
        self.plot_data['X'].append(epoch + counter_ratio)
        self.plot_data['Y'].append([losses[k] for k in self.plot_data['legend']])

        try:
            self.vis.line(

                X=np.stack([np.array(self.plot_data['X'])] * len(self.plot_data['legend']), 1),
                Y=np.array(self.plot_data['Y']),

                opts={
                    'title': self.name + ' loss over time',
                    'legend': self.plot_data['legend'],
                    'xlabel': 'epoch',
                    'ylabel': 'loss'},
                win=self.display_id)

It says that X.shape == Y.shape, 'X and Y should be the same shape',
I don't know what wrong with this, could you help clarify what is happening here?
Thanks a lot.
By the way, to train with only the L1 loss, can I just set the hyper-parameter of GAN loss as 0?

**Comment by wgh0** [comment]
Hello, I have the same problem, how do you solve it
